src/core/Class.js

```diff
@@ -8,110 +8,116 @@ import * as Util from './Util';
 
 // Thanks to John Resig and Dean Edwards for inspiration!
 
-export function Class() {}
-
-Class.extend = function (props) {
-
+export class Class {
 	// @function extend(props: Object): Function
 	// [Extends the current class](#class-inheritance) given the properties to be included.
 	// Returns a Javascript function that is a class constructor (to be called with `new`).
-	const NewClass = function (...args) {
+	static extend(props) {
+		class NewClass {
+			_initHooksCalled = false;
 
-		Util.setOptions(this);
+			constructor(...args) {
+				Util.setOptions(this);
 
-		// call the constructor
-		if (this.initialize) {
-			this.initialize.apply(this, args);
-		}
+				// call the constructor
+				if (this.initialize) {
+					this.initialize.apply(this, args);
+				}
+
+				// call all constructor hooks
+				this.callInitHooks();
+			}
 
-		// call all constructor hooks
-		this.callInitHooks();
-	};
+			callInitHooks() {
+				if (this._initHooksCalled) { return; }
 
-	const parentProto = this.prototype;
+				// collect all prototypes in chain
+				const protos = [];
+				let proto = Object.getPrototypeOf(this);
 
-	const proto = Object.create(parentProto);
-	proto.constructor = NewClass;
+				while (proto !== null) {
+					protos.push(proto);
+					proto = Object.getPrototypeOf(proto);
+				}
 
-	NewClass.prototype = proto;
+				// reverse so the parent prototype is first
+				protos.reverse();
 
-	// inherit parent's statics
-	for (const i in this) {
-		if (Object.hasOwn(this, i) && i !== 'prototype') {
-			NewClass[i] = this[i];
+				// call init hooks on each prototype
+				protos.forEach((proto) => {
+					const initHooks = proto._initHooks ?? [];
+					initHooks.forEach(hook => hook.call(this));
+				});

```

Comment:
Nit: avoid closures, prefer the more idiomatic `of`.

```suggestion
				for (const proto of protos) {
					const initHooks = proto._initHooks ?? [];
					for (const hook of initHooks) hook.call(this);
				}
```

Reply:
Done 👍